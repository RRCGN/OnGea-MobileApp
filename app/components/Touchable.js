/**
 * Helper for native Touchable Views across platforms.
 * Uses TouchableNativeFeedback in Android and TouchableOpacity in iOS.
 * @flow
 */

import React from 'react'
import {
  View,
  Platform,
  TouchableOpacity,
  TouchableNativeFeedback
} from 'react-native'


type Props = {
  onPress: ?() => void,
  onPressIn: ?() => void,
  onPressOut: ?() => void,
  rippleColor?: ?string,
  borderRadius?: number,
  useForeground?: boolean,
  children: ReactElement<*>
}

const Touchable = ({
  onPress,
  onPressIn,
  onPressOut,
  rippleColor = null,
  borderRadius = 0,
  useForeground = false,
  children
}: Props) => {

  // Use TouchableNativeFeedback for Android
  if (Platform.OS === 'android') {
    const touchable = (
      <TouchableNativeFeedback
        useForeground={TouchableNativeFeedback.canUseNativeForeground()
          ? useForeground
          : false
        }
        background={rippleColor
          ? TouchableNativeFeedback.Ripple(rippleColor, borderRadius > 0 ? true : false)
          : TouchableNativeFeedback.SelectableBackground()
        }
        disabled={!onPress}
        onPress={onPress}
        onPressIn={onPressIn}
        onPressOut={onPressOut}
      >
        {children}
      </TouchableNativeFeedback>
    )

    if (borderRadius > 0) {
      return <View style={{ borderRadius }}>{touchable}</View>
    }

    return touchable
  }

  // Use TouchableOpacity for iOS
  if (Platform.OS === 'ios') {
    return (
      <TouchableOpacity
        onPress={onPress}
        nPressIn={onPressIn}
        onPressOut={onPressOut}
      >
        <View>{children}</View>
      </TouchableOpacity>
    )
  }
}

export default Touchable
