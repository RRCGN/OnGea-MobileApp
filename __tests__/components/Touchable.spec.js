import { View } from 'react-native'
import React from 'react'
import Touchable from '../../src/components/Touchable'

import renderer from 'react-test-renderer'

describe('<Touchable />', () => {

  describe('on ios', () => {
    test('renders correctly with default props', () => {
      const tree = renderer.create(
        <Touchable><View /></Touchable>
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })

    test('renders correctly with onPress', () => {
      const tree = renderer.create(
        <Touchable onPress={() => {}}><View /></Touchable>
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

})
