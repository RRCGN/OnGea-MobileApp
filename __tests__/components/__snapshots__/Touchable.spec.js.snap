exports[`<Touchable /> on ios renders correctly with default props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <View>
    <View />
  </View>
</View>
`;

exports[`<Touchable /> on ios renders correctly with onPress 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <View>
    <View />
  </View>
</View>
`;
